lib/BridgeXPathAbstract.php

```diff
@@ -0,0 +1,565 @@
+	 * @return string
+	 */
+	protected function getExpressionItemTitle(){
+		return static::XPATH_EXPRESSION_ITEM_TITLE;
+	}
+
+	/**
+	 * XPath expression for extracting an item's content from the item context
+	 * @return string
+	 */
+	protected function getExpressionItemContent(){
+		return static::XPATH_EXPRESSION_ITEM_CONTENT;
+	}
+
+	/**
+	 * XPath expression for extracting an item link from the item context
+	 * @return string
+	 */
+	protected function getExpressionItemUri(){
+		return static::XPATH_EXPRESSION_ITEM_URI;
+	}
+
+	/**
+	 * XPath expression for extracting an item author from the item context
+	 * @return string
+	 */
+	protected function getExpressionItemAuthor(){
+		return static::XPATH_EXPRESSION_ITEM_AUTHOR;
+	}
+
+	/**
+	 * XPath expression for extracting an item timestamp from the item context
+	 * @return string
+	 */
+	protected function getExpressionItemTimestamp(){
+		return static::XPATH_EXPRESSION_ITEM_TIMESTAMP;
+	}
+
+	/**
+	 * XPath expression for extracting item enclosures (media content like
+	 * images or movies) from the item context
+	 * @return string
+	 */
+	protected function getExpressionItemEnclosures(){
+		return static::XPATH_EXPRESSION_ITEM_ENCLOSURES;
+	}
+
+	/**
+	 * XPath expression for extracting an item category from the item context
+	 * @return string
+	 */
+	protected function getExpressionItemCategories(){
+		return static::XPATH_EXPRESSION_ITEM_CATEGORIES;
+	}
+
+	/**
+	 * Fix encoding
+	 * @return string
+	 */
+	protected function getSettingFixEncoding(){
+		return static::SETTING_FIX_ENCODING;
+	}
+
+	/**
+	 * Internal helper method for quickly accessing all the user defined constants
+	 * in derived classes
+	 *
+	 * @param $name
+	 * @return bool|string
+	 */
+	private function getParam($name){
+		switch($name) {
+
+			case 'url':
+				return $this->getSourceUrl();
+			case 'feed_title':
+				return $this->getExpressionTitle();
+			case 'feed_icon':
+				return $this->getExpressionIcon();
+			case 'item':
+				return $this->getExpressionItem();
+			case 'title':
+				return $this->getExpressionItemTitle();
+			case 'content':
+				return $this->getExpressionItemContent();
+			case 'uri':
+				return $this->getExpressionItemUri();
+			case 'author':
+				return $this->getExpressionItemAuthor();
+			case 'timestamp':
+				return $this->getExpressionItemTimestamp();
+			case 'enclosures':
+				return $this->getExpressionItemEnclosures();
+			case 'categories':
+				return $this->getExpressionItemCategories();
+			case 'fix_encoding':
+				return $this->getSettingFixEncoding();
+		}
+	}
+
+	/**
+	 * Should provide the source website HTML content
+	 * can be easily overwritten for example if special headers or auth infos are required
+	 * @return string|string[]
+	 */
+	protected function provideWebsiteContent() {
+		return getContents($this->feedUri);
+	}
+
+	/**
+	 * Should provide the feeds title
+	 *
+	 * @param DOMXPath $xpath
+	 * @return string|string[]
```

Comment:
Not familiar with doc's syntax. Does it include situation, where function does not return anything?

Reply:
> 
> 
> Not familiar with doc's syntax. Does it include situation, where function does not return anything?

I think it was added in error. I removed this "doc's syntax"